storybookjs/storybook
# Docs: Exclude decorators by default from source

## What I did

Due to reordering in #21182 the source decorator comes *before* any user decorator in `preview.js`. This means that we commonly trigger the bug in #21649.

Really it was a bug that was fixed by #21182 that the user's decorators *weren't* visible in the source by default given we weren't excluding them.

So it makes more sense to default this to true and people can opt out in cases where they want to show the decorator.



## How to test

Add the following to a React sandbox's `preview.tsx`:

```tsx
  decorators: [
    (S) => <div><S/></div>
  ]
```

Then check the source block of a story:

<img width="418" alt="image" src="https://user-images.githubusercontent.com/132554/226782350-36eefdfe-b9c5-4f1d-bc0b-f4eea73d7ad7.png">


## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
